content/opt/setup/30-locales

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+
+if [[ ! -f /etc/locale.gen.dist ]]
+then
+   cp /etc/locale.gen /etc/locale.gen.dist
+fi
+
+echo "BUILD_LOCALES: $BUILD_LOCALES"
+typeset -i m=$(echo $BUILD_LOCALES | awk -F ',' '{print NF}')
+typeset -i n=1
+while [[ $n -le $m ]]
+do
+   b_locale=$(echo $BUILD_LOCALES | awk -F ',' "{print \$$n}")
+   cat /etc/locale.gen |grep -q "^$b_locale"
+   if [[ $? -ne 0 ]]
+   then
+      echo "adding locale $b_locale"
+      cat /etc/locale.gen | sed "s/# $b_locale/$b_locale/" > /tmp/locale.gen
```

Comment:
If the locale, I specified is not there in the file, it won't get enabled. On the one hand side, this is great. But KIM, that you could have a typo in the environment variable (e.g.: using `en_us.UTF-8 UTF-8`) and then the actual locale gets silently dropped.

---

`sed` has got a `-i` switch. And also like `grep` it takes a filename as the last parameters.

```
sed -i "s/# $b_locale/$b_locale/" /etc/locale.gen
```